x-pack/plugins/security/server/routes/user_profile/bulk_get.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import type { RouteDefinitionParams } from '..';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+export function defineBulkGetUserProfilesRoute({
+  router,
+  getUserProfileService,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/user_profile/_bulk_get',
+      validate: {
+        body: schema.object({
+          uids: schema.arrayOf(schema.string(), { minSize: 1 }),
+          data: schema.maybe(schema.string()),
+        }),
+      },
+      options: { tags: ['access:bulkGetUserProfiles'] },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      const userProfileService = getUserProfileService();
```

Comment:
This is actually the internal user profile service so might be worth renaming this function to clarify?

Reply:
Discussed offline - changed to `const userProfileServiceInternal = getUserProfileService();`.